dashboard/client/src/components/NewEventTable.tsx

```diff
@@ -0,0 +1,319 @@
+  Tooltip,
+} from "@material-ui/core";
+import { SearchOutlined } from "@material-ui/icons";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+import Pagination from "@material-ui/lab/Pagination";
+import dayjs from "dayjs";
+import React, { useState } from "react";
+import { Align, Filters } from "../type/event";
+
+import { useFilter } from "../util/hook";
+import { SeverityLevel } from "./event";
+import Loading from "./Loading";
+import { StatusChip } from "./StatusChip";
+import { useEvents } from "./useEvents";
+
+type EventTableProps = {
+  defaultSeverityLevels?: SeverityLevel[];
+  entityName?: string;
+  entityId?: string; // It could be a specific or "*" to represent all entities
+};
+
+const useStyles = makeStyles((theme) => ({
+  overflowCell: {
+    display: "block",
+    margin: "auto",
+    maxWidth: 360,
+    textOverflow: "ellipsis",
+    overflow: "hidden",
+    whiteSpace: "nowrap",
+  },
+  filterContainer: {
+    display: "flex",
+    alignItems: "center",
+  },
+  search: {
+    margin: theme.spacing(1),
+    display: "inline-block",
+    fontSize: 12,
+    lineHeight: "46px",
+    height: 56,
+  },
+
+  tableContainer: {
+    overflowX: "scroll",
+  },
+
+  helpInfo: {
+    marginLeft: theme.spacing(1),
+  },
+  message: {
+    maxWidth: 200,
+  },
+  pagination: {
+    marginTop: theme.spacing(3),
+  },
+  root: {
+    height: 800,
+    paddingLeft: theme.spacing(1),
+  },
+}));
+
+const SOURCE_TYPE_OPTIONS = [
+  "COMMON",
+  "CORE_WORKER",
+  "GCS",
+  "RAYLET",
+  "JOBS",
+  "SERVE",
+  "CLUSTER_LIFECYCLE",
+  "AUTOSCALER",
+];
+
+const SEVERITY_LEVEL_OPTIONS = ["INFO", "DEBUG", "WARNING", "ERROR", "TRACING"];
+
+const COLUMNS = [
+  { label: "Severity", align: "center" },
+  { label: "Timestamp", align: "center" },
+  { label: "Source", align: "center" },
+  { label: "Custom fields", align: "left" },
+  { label: "Message", align: "left" },
+];
+
+const useEventTable = (props: EventTableProps) => {
+  const { defaultSeverityLevels, entityName, entityId } = props;
+  const { changeFilter: _changeFilter, filterFunc } = useFilter();
+  const [filters, _setFilters] = useState<Filters>({
+    sourceType: [],
+    severityLevel: defaultSeverityLevels || [],
+    entityName, // We used two fields(entityName, entityId) because we will support select entityName by dropdown and input entityId by TextField in the future.
+    entityId, // id or *
+  });
+
+  const [pagination, setPagination] = useState({
+    pageNo: 1, // first page is PageNo 1
+    pageSize: 10,
+  });
+
+  const changePage = (key: string, value: number) => {
+    setPagination({ ...pagination, [key]: value });
+  };
+
+  const changeFilter: typeof _changeFilter = (...params) => {
+    _changeFilter(...params);
+    setPagination({
+      ...pagination,
+      pageNo: 1,
+    });
+  };
+
+  const setFilters: typeof _setFilters = (...params) => {
+    _setFilters(...params);
+    setPagination({
+      ...pagination,
+      pageNo: 1,
+    });
+  };
```

Comment:
Can you try wrapping these in `useCallback`, this could prevent unnecessary re-renders and possibly solve your filter input UI issue?